src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java

```diff
@@ -96,10 +101,26 @@ public boolean isLightweight() {
         }
         Run<?,?> build = (Run<?,?>) _build;
         if (isLightweight()) {
-            try (SCMFileSystem fs = SCMFileSystem.of(build.getParent(), scm)) {
+            try (SCMFileSystem fs = SCMFileSystem.of(build.getParent(), scm, SCMRevisionAction.getRevision(build))) {
```

Comment:
Technically unrelated, but signals a bug waiting to happen... if there are commits concurrent with the build, we could end up checking out the wrong (newer) revision of the `Jenkinsfile` than the checkout will check out if we do not at least attempt to resolve the revision

Reply:
`CpsScmFlowDefintion` does not support exact revision matching between `Jenkinsfile` and `checkout scm`. It _could_, if using `scm-api` (there is a filed RFE for it), but such a change does not belong here I think; should be done separately, with test coverage.

Reply:
[JENKINS-36453](https://issues.jenkins-ci.org/browse/JENKINS-36453) FTR

Reply:
Well if you attach an `SCMRevisionAction` to the build in this step then you get [JENKINS-36453](https://issues.jenkins-ci.org/browse/JENKINS-36453) "for free"... but as this PR is mostly a PoC to show what should be done...